Sources/PrimerSDK/Classes/Data Models/PrimerInternalSessionFlow.swift

```diff
@@ -46,49 +45,72 @@ public enum PrimerSessionFlow {
     case addDirectDebit
     case checkoutWithKlarna
 
-    var vaulted: Bool {
+    internal var internalSessionFlow: PrimerInternalSessionFlow {
         switch self {
-        case .addCardToVault:
-            return true
-        case .addPayPalToVault:
-            return true
         case .default:
-            return false
-        case .addDirectDebit:
-            return true
+            return .checkout
+        case .defaultWithVault:
+            return .vault
         case .completeDirectCheckout:
-            return false
-        case .checkoutWithKlarna:
-            return false
+            return .checkoutWithCard
+        case .addPayPalToVault:
+            return .vaultPayPal
+        case .addCardToVault:
+            return .vaultCard
         case .addDirectDebitToVault:
-            return true
+            return .vaultDirectDebit
         case .addKlarnaToVault:
+            return .vaultKlarna
+        case .addDirectDebit:
+            return .vaultDirectDebit
+        case .checkoutWithKlarna:
+            return .checkoutWithKlarna
+        }
+    }
+}
+
+internal enum PrimerInternalSessionFlow {
```

Comment:
is this so we dont expose the singleton to the developer?

Reply:
This isn't a singleton, just an enum. This is so we can use naming that makes more sense internally, while not making breaking changes.